### PR TITLE
fix: sanitizing name/email before submitting it to backend

### LINKED ISSUE #258: Invalid username
Subject: invalid username
Description:
Whatever username i try to sign up with it just says invalid username

Steps:
Sign up as a new user

Results:
Says invalid username for any username i try

Backend version: bf2ff87
Frontend version: bf2ff87
User Agent: Mozilla/5.0 (Android 6.0.1; Mobile; rv:68.0) Gecko/68.0 Firefox/68.0
Page: https://app.couchers.org/signup/Ft_mArU2qVsbbXPAMtVMb4o8qWncLBNky3QsxwQkeO4=
User ID: 0

### PR DESCRIPTION
**Pull request by Tay0108**
Closes https://github.com/Couchers-org/couchers/issues/258

Let me know if there are other places we should do this. Also this function is used for sanitizing both email and login, couldn't come up with a better name, feel free to suggest sth better ;) 

Do we display username somewhere? Does it make sense to display it the way user entered it during sign up (capitalized or not)?

### CODE REVIEW

File: app/frontend/src/features/auth/email/ChangeEmailPage.tsx [hunk 1]
```diff
@@ -29,7 +30,8 @@ export default function ChangeEmailPage() {
     reset: resetForm,
   } = useForm<ChangeEmailFormData>();
   const onSubmit = handleSubmit(({ currentPassword, newEmail }) => {
```
Comment (darrenvong) [inline app/frontend/src/features/auth/email/ChangeEmailPage.tsx:32]: Would be nice to update and add to the existing tests on this page to check it's the sanitised input that's being called.

**Review by darrenvong — CHANGES_REQUESTED** [review]
I think apart from the URL, personally not aware of the username being displayed anywhere else. Is that right @lucaslcode ?

File: app/frontend/src/features/auth/email/ChangeEmailPage.tsx (hunk 1 above)
Comment (Tay0108) [reply app/frontend/src/features/auth/email/ChangeEmailPage.tsx:32]: I created 2 new tests for password and without password forms. If you'd rather have me edit existing tests let me know ;)

**Review by lucaslcode — APPROVED** [review]

File: app/frontend/src/features/auth/email/ChangeEmailPage.tsx (hunk 1 above)
Comment (darrenvong) [reply app/frontend/src/features/auth/email/ChangeEmailPage.tsx:32]: Haha more _useful_ tests are always welcomed, but don't want to blow the scope of this too much for ones that don't have tests yet. This tests the helper as well which I much prefer than unit testing it in this case as it's only a thin wrapper around built-in functionality so we know that should work on its own already.

**Review by darrenvong — APPROVED** [review]

**Pull request merged**